libs/s25main/EconomyModeHandler.cpp

```diff
@@ -22,181 +22,178 @@
 #include "GamePlayer.h"
 #include "GlobalGameSettings.h"
 #include "SerializedGameData.h"
+#include "helpers/containerUtils.h"
 #include "random/Random.h"
 #include "world/GameWorld.h"
 #include "world/GameWorldGame.h"
 #include "gameTypes/JobTypes.h"
 #include "gameData/JobConsts.h"
 #include <boost/optional.hpp>
 
-auto getPlayerMask = [](unsigned playerId) { return 1u << playerId; };
-
-GoodType types_less[]{
-  /*  1 */ GD_TONGS,      // Zange
-  /*  2 */ GD_HAMMER,     // Hammer
-  /*  3 */ GD_AXE,        // Axt
-  /*  4 */ GD_SAW,        // Saege
-  /*  5 */ GD_PICKAXE,    // Spitzhacke
-  /*  6 */ GD_SHOVEL,     // Schaufel
-  /*  7 */ GD_CRUCIBLE,   // Schmelztiegel
-  /*  8 */ GD_RODANDLINE, // Angel
-  /*  9 */ GD_SCYTHE,     // Sense
-  /* 12 */ GD_CLEAVER,    // Beil
-  /* 13 */ GD_ROLLINGPIN, // Nudelholz
-  /* 14 */ GD_BOW,        // Bogen
+std::array<GoodType, 12> specialGoodPool = {
+  /*  1 */ GD_TONGS,
+  /*  2 */ GD_HAMMER,
+  /*  3 */ GD_AXE,
+  /*  4 */ GD_SAW,
+  /*  5 */ GD_PICKAXE,
+  /*  6 */ GD_SHOVEL,
+  /*  7 */ GD_CRUCIBLE,
+  /*  8 */ GD_RODANDLINE,
+  /*  9 */ GD_SCYTHE,
+  /* 12 */ GD_CLEAVER,
+  /* 13 */ GD_ROLLINGPIN,
+  /* 14 */ GD_BOW,
 };
 
-GoodType types_more[]{
-  /*  0 */ GD_BEER,    // Bier
-  /* 11 */ GD_WATER,   // Wasser
-  /* 15 */ GD_BOAT,    // Boot
-  /* 16 */ GD_SWORD,   // Schwert
-  /* 17 */ GD_IRON,    // Eisen
-  /* 18 */ GD_FLOUR,   // Mehl
-  /* 19 */ GD_FISH,    // Fisch
-  /* 20 */ GD_BREAD,   // Brot
-  /* 22 */ GD_WOOD,    // Holz
-  /* 23 */ GD_BOARDS,  // Bretter
-  /* 24 */ GD_STONES,  // Steine
-  /* 27 */ GD_GRAIN,   // Getreide
-  /* 28 */ GD_COINS,   // Mnzen
-  /* 29 */ GD_GOLD,    // Gold
-  /* 30 */ GD_IRONORE, // Eisenerz
-  /* 31 */ GD_COAL,    // Kohle
-  /* 32 */ GD_MEAT,    // Fleisch
-  /* 33 */ GD_HAM,     // Schinken ( Schwein )
+std::array<GoodType, 18> commonGoodPool = {
+  /*  0 */ GD_BEER,
+  /* 11 */ GD_WATER,
+  /* 15 */ GD_BOAT,
+  /* 16 */ GD_SWORD,
+  /* 17 */ GD_IRON,
+  /* 18 */ GD_FLOUR,
+  /* 19 */ GD_FISH,
+  /* 20 */ GD_BREAD,
+  /* 22 */ GD_WOOD,
+  /* 23 */ GD_BOARDS,
+  /* 24 */ GD_STONES,
+  /* 27 */ GD_GRAIN,
+  /* 28 */ GD_COINS,
+  /* 29 */ GD_GOLD,
+  /* 30 */ GD_IRONORE,
+  /* 31 */ GD_COAL,
+  /* 32 */ GD_MEAT,
+  /* 33 */ GD_HAM,
 };
 
-EconomyModeHandler::EconomyModeHandler(unsigned end_frame) : end_frame(end_frame), last_updated(0)
+const unsigned int numGoodTypesToCollect = 7;
```

Comment:
`constexpr unsigned numGoodTypesToCollect = 7;` (remove the int for consistency and use constexpr where possible